flake.nix

```diff
@@ -138,18 +134,18 @@
         apps = {
           default ={
             type = "app";
-            program = "${self.packages.${targetSystem}.application}/bin/polykey";
+            program = "${self.packages.${targetSystem}.default}/bin/polykey";
           };
-          polykey-cli = {
+          executable = {
             type = "app";
-            program = "${self.packages.${targetSystem}.polykey-cli}";
+            program = "${self.packages.${targetSystem}.executable}";
           };
         };
 
         packages = {
-          polykey-cli = polykey-cli-pkg;
-          application = polykey-cli-app;
-          docker = dockerImage;
+          default = polykey-cli;
+          executable = polykey-cli-executable;
+          docker = docker;
```

Comment:
This most closely aligns to our original `default.nix` and such. And yes the flake system is definitely much easier! Great work @brynblack and @tegefaulkes. Now running it is much easier than remember all the `nix-build` commands. I guess the only thing is that there's no parameterisation... That's interesting.

Reply:
`nix build` does have an `--argstr` parameter but it's not exactly clear how it works. Maybe parameters are allowed?

Reply:
No, there are no parameters that can be passed. Flakes as I have said before are reproducible and sandboxed, and if you saw my Excalidraw then this would make more sense. This was the case with `npmDepsHash`, as we could not dynamically just put it in at runtime.

There does exist a way to pass parameters, but that would mean making the flakes impure, which is considered a very bad practice as now there is a possible dependency on an arbitrary parameter for a desired output. I think its best if we keep it for now with how its done, we can always build specialisations of a derivative through hard coded parameters, and then later select that derivative. I think there also exists a way to add properties and attributes to an output, so it may be possible to do more there.